server/base/src/main/java/org/apache/accumulo/server/AbstractServer.java

```diff
@@ -44,6 +48,8 @@ public abstract class AbstractServer implements AutoCloseable, MetricsProducer,
   private final ProcessMetrics processMetrics;
   protected final long idleReportingPeriodNanos;
   private volatile long idlePeriodStartNanos = 0L;
+  private volatile Thread serverThread;
+  private volatile Thread verificationThread;
```

Comment:
Do these need to be volatile? I'm not sure what that's getting us here. Should these be final and the Thread created in the constructor, and only run in the runServer method?

Reply:
I'm assuming so, but honestly not 100% sure. `serverThread.isAlive` is being called from the `verificationThread` and `verificationThread` is checked for a null reference in the main thread when stopping.